packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -37,9 +36,86 @@ type StringNotation<TSchemaUID extends Common.UID.Schema> =
  * type C = ['populatableField']; // ✅
  * type D = ['<random_string>']; // ❌
  */
-type ArrayNotation<TSchemaUID extends Common.UID.Schema> = StringNotation<TSchemaUID>[];
+export type ArrayNotation<TSchemaUID extends Common.UID.Schema> = StringNotation<TSchemaUID>[];
+
+type GetPopulatableKeysWithTarget<TSchemaUID extends Common.UID.Schema> = Extract<
+  Attribute.GetPopulatableKeys<TSchemaUID>,
+  Attribute.GetKeysWithTarget<TSchemaUID>
+>;
+
+type GetPopulatableKeysWithoutTarget<TSchemaUID extends Common.UID.Schema> = Exclude<
+  Attribute.GetPopulatableKeys<TSchemaUID>,
+  GetPopulatableKeysWithTarget<TSchemaUID>
+>;
+
+/**
+ * Fragment populate notation for polymorphic attributes
+ */
+type PopulateFragment<TMaybeTargets extends Common.UID.Schema> = {
+  on?: { [TSchemaUID in TMaybeTargets]?: boolean | Params.For<TSchemaUID> };
+};
+
+type PopulateClause<
+  TSchemaUID extends Common.UID.Schema,
+  TKeys extends Attribute.GetPopulatableKeys<TSchemaUID>
+> = {
+  [TKey in TKeys]?: boolean | Params.For<Attribute.GetTarget<TSchemaUID, TKey>>;
+};
+
+/**
+ * Object notation for populate
+ *
+ * @example
+ * type A = { image: true }; // ✅
+ * type B = { image: { fields: ['url', 'provider'] } }; // ✅
+ * type C = { populatableField: { populate: { nestedPopulatableField: true } } }; // ✅
+ * type D = { dynamic_zone: { on: { comp_A: { fields: ['name', 'price_a'] }, comp_B: { fields: ['name', 'price_b'] } } } }; // ✅
```

Comment:
I was under the impression we supported (though deprecated) something like `{ dynamic_zone: { comp_A: true } }` but I don't see it in the documentation. Can you investigate, since I think you wrote most of that code, and confirm we will only support the `on` syntax for dynamic zones?

Reply:
We support either
- `{ dynamic_zone: { on: { comp_A: true } } }`
- `{ dynamic_zone: true }`
- `{ dynamic_zone: { populate: true } }`
But not `{ dynamic_zone: { comp_A: true } }`